src/runtime/threading_backend.cc

```diff
@@ -133,22 +133,39 @@ class ThreadGroup::Impl {
           sizeof(cpu_set_t), &cpuset);
 #endif
     }
-    if (exclude_worker0) {  // bind the master thread to core 0
-      cpu_set_t cpuset;
-      CPU_ZERO(&cpuset);
-      if (reverse) {
-        CPU_SET(sorted_order_[sorted_order_.size() - 1], &cpuset);
-      } else {
-        CPU_SET(sorted_order_[0], &cpuset);
+    if (exclude_worker0) {  // master thread run task
+#if defined(__ANDROID__)
+      SetFullCpuAffinity();
+#else
+      const char* bind_master_thread = getenv("TVM_BIND_MASTER_THREAD");
+      if (bind_master_thread) {
```

Comment:
You don't really bind the master thread, `CPU_SET` only sets the flag, it will need `pthread_setaffinity_np` to take action.

Reply:
if i don't understand wrong, what you mean
```cpp
   pthread_atfork(nullptr, nullptr, ThreadGroup::Impl::SetFullCpuAffinity);
```
we only need this.

Reply:
`pthread_atfork` only take effect when a fork happens. And when it happens, the child process will run `ThreadGroup::Impl::SetFullCpuAffinity`, while the parent process does nothing (as you have `nullptr`).

Reply:
And in the `if` statement, the code doesn't actually bind the threads, but only sets the flag.

Reply:
We will need to call pthread_setaffinity_np to set the thread affinity to the cpuset here

Reply:
Understood.
```cpp
      const char* bind_master_thread = getenv("TVM_BIND_MASTER_THREAD");
      if (bind_master_thread && atoi(bind_master_thread) == 1) {
        cpu_set_t cpuset;
        CPU_ZERO(&cpuset);
        if (reverse) {
          CPU_SET(sorted_order_[sorted_order_.size() - 1], &cpuset);
        } else {
          CPU_SET(sorted_order_[0], &cpuset);
        }
        pthread_setaffinity_np(pthread_self(), sizeof(cpu_set_t), &cpuset);
      }
      pthread_atfork(nullptr, nullptr, ThreadGroup::Impl::SetFullCpuAffinity);
```